benchmarking-cli/src/utils_input.rs

```diff
@@ -0,0 +1,264 @@
+                            format!("failed to convert advice map value `{v}` to Felt - {e}")
+                        })
+                    })
+                    .collect::<Result<Vec<_>, _>>()?;
+                Ok((key, values))
+            })
+            .collect::<Result<BTreeMap<[u8; 32], Vec<Felt>>, String>>()?;
+
+        Ok(Some(map))
+    }
+
+    /// Parse merkle store data from the input file.
+    fn parse_merkle_store(&self) -> Result<Option<MerkleStore>, String> {
+        let merkle_data = match &self.merkle_store {
+            Some(merkle_data) => merkle_data,
+            None => return Ok(None),
+        };
+
+        let mut merkle_store = MerkleStore::default();
+        for data in merkle_data {
+            match data {
+                MerkleData::MerkleTree(data) => {
+                    let leaves = Self::parse_merkle_tree(data)?;
+                    let merkle_tree = MerkleTree::new(leaves)
+                        .map_err(|e| format!("failed to add merkle tree to merkle store - {e}"))?;
+                    merkle_store.extend(merkle_tree.inner_nodes());
+                }
+                MerkleData::SparseMerkleTree(data) => {
+                    let entries = Self::parse_sparse_merkle_tree(data)?;
+                    // TODO: Support variable depth
+                    let smt =
+                        SimpleSmt::with_leaves(SimpleSmt::MAX_DEPTH, entries).map_err(|e| {
+                            format!("failed to add sparse merkle tree to merkle store - {e}")
+                        })?;
+                    merkle_store.extend(smt.inner_nodes());
+                }
+            }
+        }
+
+        Ok(Some(merkle_store))
+    }
+
+    /// Parse and return merkle tree leaves.
+    fn parse_merkle_tree(tree: &[String]) -> Result<Vec<Word>, String> {
+        tree.iter()
+            .map(|v| {
+                let leaf = Self::parse_word(v)?;
+                Ok(leaf)
+            })
+            .collect()
+    }
+
+    /// Parse and return sparse merkle tree entries.
+    fn parse_sparse_merkle_tree(tree: &[(u64, String)]) -> Result<Vec<(u64, Word)>, String> {
+        tree.iter()
+            .map(|(index, v)| {
+                let leaf = Self::parse_word(v)?;
+                Ok((*index, leaf))
+            })
+            .collect()
+    }
+
+    /// Parse a `Word` from a hex string.
+    pub fn parse_word(word_hex: &str) -> Result<Word, String> {
+        let mut word_data = [0u8; 32];
+        hex::decode_to_slice(word_hex, &mut word_data)
+            .map_err(|e| format!("failed to decode `Word` from hex {word_hex} - {e}"))?;
+        let mut word = Word::default();
+        for (i, value) in word_data.chunks(8).enumerate() {
+            word[i] = Felt::try_from(value).map_err(|e| {
+                format!("failed to convert `Word` data {word_hex} (element {i}) to Felt - {e}")
+            })?;
+        }
+        Ok(word)
+    }
```

Comment:
A few changes:

- This should handle the case were `word_hex` has an invalid number of bytes (either too many or too little).
- These methods that are not using `self`, are not constructors, and are not coming from a trait, should IMO be standalone functions.

---

Nit: It would also be nice to have this on `crypto` repo, along side this:

https://github.com/0xPolygonMiden/crypto/blob/18302d68e012fb008b5318da47e1525e219b97a8/src/utils/mod.rs#L24-L36

(probably better to open a separate PR and not block this one on that)

Reply:
> This should handle the case were word_hex has an invalid number of bytes (either too many or too little).

Will be done in the common file, actually we should use this here 

https://github.com/0xPolygonMiden/miden-vm/blob/5c5749549d5cf1ccb4a6b9c5cf38427a41537145/miden/src/cli/data.rs#L268-L280